src/dotnet/CarbonAware.Aggregators/src/CarbonAware/CarbonAwareAggregator.cs

```diff
@@ -9,18 +9,44 @@ public class CarbonAwareAggregator : ICarbonAwareAggregator
     {
         private readonly ILogger<CarbonAwareAggregator> _logger;
 
-        private readonly ICarbonAware _plugin;
+        private readonly ICarbonIntensityDataSource _dataSource;
 
-        public CarbonAwareAggregator(ILogger<CarbonAwareAggregator> logger, ICarbonAware plugin)
+        public CarbonAwareAggregator(ILogger<CarbonAwareAggregator> logger, ICarbonIntensityDataSource dataSource)
         {
             this._logger = logger ?? throw new ArgumentNullException(nameof(logger));
-            this._plugin = plugin;
+            this._dataSource = dataSource;
         }
 
         public async Task<IEnumerable<EmissionsData>> GetEmissionsDataAsync(IDictionary props)
         {
-            return await this._plugin.GetEmissionsDataAsync(props);
+            return await this._dataSource.GetCarbonIntensityAsync(GetOrDefaultLocation(props), GetDefaultPeriodOrThrow(props), GetDefaultPeriodOrThrow(props, false));
 
         }
+
+        private DateTimeOffset GetDefaultPeriodOrThrow(IDictionary props, bool isStart = true) {
+            if (isStart) {
+                var start = props[CarbonAwareConstants.Start];
```

Comment:
Seems like you can simplify by having the DateTimeOffset outside of the if/else. Just return `var date`

Reply:
Changed the logic a bit to cleanup